### PR TITLE
Adding Local Directory type

### PR DESCRIPTION
**Pull request by APriestman**

### CODE REVIEW

File: bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java [hunk 1]
```diff
@@ -3890,6 +3890,162 @@ public VirtualDirectory addVirtualDirectory(long parentId, String directoryName,
...
+			if (parentPath == null) {
+				parentPath = "/"; //NON-NLS
+			}
+			String parentName = getFileName(parentId, connection);
+			if (parentName != null && !parentName.isEmpty()) {
+				parentPath = parentPath + parentName + "/"; //NON-NLS
+			}
+
+			// Insert a row for the virtual directory into the tsk_objects table.
+			// INSERT INTO tsk_objects (par_obj_id, type) VALUES (?, ?)
```
Comment (rcordovano) [inline bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java]: Copy paste "typo"...

File: bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java [hunk 2]
```diff
@@ -3890,6 +3890,162 @@ public VirtualDirectory addVirtualDirectory(long parentId, String directoryName,
...
+				statement.setNull(1, java.sql.Types.BIGINT);
+			}
+			statement.setInt(2, TskData.ObjectType.ABSTRACTFILE.getObjectType());
+			connection.executeUpdate(statement);
+			resultSet = statement.getGeneratedKeys();
+			resultSet.next();
+			long newObjId = resultSet.getLong(1); //last_insert_rowid()
+
+			// Insert a row for the virtual directory into the tsk_files table.
+			// INSERT INTO tsk_files (obj_id, fs_obj_id, name, type, has_path, dir_type, meta_type,
```
Comment (rcordovano) [inline bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java]: Typo...

File: bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java [hunk 3]
```diff
@@ -3890,6 +3890,162 @@ public VirtualDirectory addVirtualDirectory(long parentId, String directoryName,
...
+				long parentFs = this.getFileSystemId(parentId, connection);
+				if (parentFs != -1) {
+					statement.setLong(2, parentFs);
+				} else {
+					statement.setNull(2, java.sql.Types.BIGINT);
+				}
+			} else {
+				statement.setNull(2, java.sql.Types.BIGINT);
+			}
+
```
Comment (rcordovano) [inline bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java]: Is it true that a LocalDirectory is never part of a file system?

File: bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java [hunk 4]
```diff
@@ -3890,6 +3890,162 @@ public VirtualDirectory addVirtualDirectory(long parentId, String directoryName,
...
+			statement.setString(15, parentPath);
+
+			// data source object id (same as object id if this is a data source)
+			long dataSourceObjectId;
+			if (0 == parentId) {
+				dataSourceObjectId = newObjId;
+			} else {
+				dataSourceObjectId = getDataSourceObjectId(connection, parentId);
+			}
+			statement.setLong(16, dataSourceObjectId);
```
Comment (rcordovano) [inline bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java]: Is it true that a LocalDirectory can never be a data source?

File: bindings/java/src/org/sleuthkit/datamodel/SpecialDirectory.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+import java.util.List;
+import org.sleuthkit.datamodel.TskData.FileKnown;
+import org.sleuthkit.datamodel.TskData.TSK_FS_META_TYPE_ENUM;
+import org.sleuthkit.datamodel.TskData.TSK_FS_NAME_FLAG_ENUM;
+import org.sleuthkit.datamodel.TskData.TSK_FS_NAME_TYPE_ENUM;
+
+/**
+ * Parent class for special directory types (Local and Virtual)
+ */
+public abstract class SpecialDirectory extends AbstractFile {
```
Comment (rcordovano) [inline bindings/java/src/org/sleuthkit/datamodel/SpecialDirectory.java]: Can access to this class be restricted to the package?

Comment (APriestman) [inline bindings/java/src/org/sleuthkit/datamodel/SpecialDirectory.java]: SpecialDirectoryNode uses it.

**Pull request merged**